### DISCUSSION TITLE
Prevent browser from opening when running npx expo start --web

### DISCUSSION THREAD

**Discussion by mathias-berg** [expo/expo#21595] [Archived: Questions] [answered]
Hi!

I'm building an Electron app (and a mobile version later on) using Expo and React Native Web.
Is there some way you can prevent the browser from opening when running `npx expo start --web` (with Webpack as a bundler)?
Since I am doing Electron I am using some Node APIs so there will be nothing showing in the opened browser anyway, just an unnecessarily step.

If I use Metro as a bundler instead and run `npx expo start` the Metro bundler is ready with a port so I can just run Electron and then everything will bundle and just the Electron window will open.

So why not use Metro as a bundler you say? When using Electron in production the baseUrl must be `./` instead of `/` since it is file based and I can't seem to find an option in Metro to change that, in Webpack that is just a simple change in the webpack config file.

Thanks in advance / Mathias

**Comment by jarredkenny** [comment]
Would also love to know this. If there is a way to disable it I have not found it and this is super annoying behaviour

**Comment by jroush-ipg** [comment]
I would also like to know this.  I've just encountered an expo project and would like to try running the dev web server in a docker container, but it keeps crashing b/c it can't launch a web browser instead.  I already know how to launch a web browser!

**Comment by mathias-berg** [comment]
I just went with a custom Webpack config instead so now I can start my server without the browser starting (not using npc expo start --web anymore). I'm still using Expo but I start every process by itself instead with one command.

**Comment by jroush-ipg** [comment, marked as answer]
After investigating, it looks like launching the browser is done by `create-react-app`, and not expo itself.  It can be disabled by setting an environment variable:
```
export BROWSER=none
```

create-react-app config docs: https://create-react-app.dev/docs/advanced-configuration
see also: https://github.com/facebook/create-react-app/issues/3070#issuecomment-327217282

**Comment by AlexGrafe** [comment]
Add a `.env` file in your project root with content
```
BROWSER=none
```

Thanks @jroush-ipg for this. If you follow the docs further down the rabbit hole, you'll find that you can put any environment variables in an `.env` file instead of your global shell environment – which I would recommend.

.env documentation for create-react-app: https://create-react-app.dev/docs/adding-custom-environment-variables/#adding-development-environment-variables-in-env

**Reply by ex3ndr** [reply]
This worked for me too!

**Comment by oliseonwu** [comment]
For me, I specified the platform I am building in app.json by adding this property -> "platforms": ["ios", "android"],
